File: clients/go/examples/collections-create-collection-lexical.go
package main

import (
	"context"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get an existing database
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	// Create a collection
	_, err := database.CreateCollection(
		ctx,
		"**COLLECTION_NAME**",
		options.CreateCollection().
			UpdateLexical(options.Lexical().
				SetEnabled(true).
				SetCustomAnalyzer(map[string]any{
					"tokenizer": map[string]any{
						"name": "standard",
						"args": map[string]any{},
					},
					"filters": []map[string]any{
						{"name": "lowercase"},
						{"name": "stop"},
						{"name": "porterstem"},
						{"name": "asciifolding"},
					},
					"charFilters": []map[string]any{},
				})),
	)
	if err != nil {
		log.Fatal(err)
	}
}

File: clients/java/examples/collections-create-collection-lexical.java
import static com.datastax.astra.client.core.lexical.AnalyzerTypes.STANDARD;

import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.definition.CollectionDefinition;
import com.datastax.astra.client.core.lexical.Analyzer;
import com.datastax.astra.client.core.lexical.LexicalOptions;
import com.datastax.astra.client.databases.Database;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    database.createCollection(
        "**COLLECTION_NAME**",
        new CollectionDefinition()
            .lexical(
                new LexicalOptions()
                    .enabled(true)
                    .analyzer(
                        new Analyzer()
                            .tokenizer(STANDARD.getValue())
                            .addFilter("lowercase")
                            .addFilter("stop")
                            .addFilter("porterstem")
                            .addFilter("asciifolding"))));
  }
}

// ==============  BOUNDARY BETWEEN EXAMPLE SNIPPETS  ==============

import static com.datastax.astra.client.core.lexical.AnalyzerTypes.STANDARD;

import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.definition.CollectionDefinition;
import com.datastax.astra.client.core.lexical.Analyzer;
import com.datastax.astra.client.core.lexical.LexicalOptions;
import com.datastax.astra.client.databases.Database;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    // Create a collection
    CollectionDefinition collectionDefinition = new CollectionDefinition();

    // Lexical Options
    Analyzer analyzer =
        new Analyzer()
            .tokenizer(STANDARD.getValue())
            .addFilter("lowercase")
            .addFilter("stop")
            .addFilter("porterstem")
            .addFilter("asciifolding");
    LexicalOptions lexicalOptions = new LexicalOptions().enabled(true).analyzer(analyzer);
    collectionDefinition.lexical(lexicalOptions);

    database.createCollection("**COLLECTION_NAME**", collectionDefinition);
  }
}
